From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 27 Jun 2022 11:54:37 -0400
Subject: [PATCH] redpitaya: For Windows, target Vista or newer for winsock2
 features.

The redpitaya source and sink use inet_pton, which on Windows when using
winsock2 is only defined when targeting Windows Vista or newer. This
sets that target by adding a definition for `_WIN32_WINNT=0x0600`.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 lib/redpitaya/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/redpitaya/CMakeLists.txt b/lib/redpitaya/CMakeLists.txt
index 78f816b..7cfac7d 100644
--- a/lib/redpitaya/CMakeLists.txt
+++ b/lib/redpitaya/CMakeLists.txt
@@ -25,6 +25,13 @@ target_include_directories(gnuradio-osmosdr PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}
 )
 
+if(MSVC)
+    # winsock2's inet_pton only defined for Vista and newer
+    target_compile_definitions(gnuradio-osmosdr PRIVATE
+        -D_WIN32_WINNT=0x0600
+    )
+endif(MSVC)
+
 APPEND_LIB_LIST(
     ${Gnuradio-blocks_LIBRARIES}
 )
-- 
2.35.1

